src/librustc_mir/transform/const_prop.rs

```diff
@@ -334,7 +343,8 @@ impl<'b, 'a, 'tcx:'b> ConstPropagator<'b, 'a, 'tcx> {
                             }
                             Value::ByVal(val)
                         };
-                        Some((val, place_ty, span))
+
+                        Some((const_value_to_alloc(self.tcx, val, place_ty), place_ty, span))
```

Comment:
His interns an intermediate/temp value. Won't that slowly fill up memory without really using it